app/models/user.rb

```diff
@@ -84,6 +87,33 @@ class User < ApplicationRecord
 
   attr_accessor :invite_code
 
+  def pam_conflict(_)
+    # block pam login tries on traditional account
+    nil
+  end
+
+  def pam_conflict?
+    return false unless Devise.pam_authentication
+    encrypted_password.present? && is_pam_account?
+  end
+
+  def pam_get_name
+    return account.username if account.present?
+    super
+  end
+
+  def pam_setup(_attributes)
+    acc = Account.new(username: pam_get_name)
+    acc.save!(validate: false)
```

Comment:
I'm not entirely sure `validate: false` works on the `!` version of the save method, I have a vague memory of running into this. More importantly, why do you need to not validate the account? I'm pretty sure the username is the *only* required part, and you do give it here.